APMrover2/mode.cpp

```diff
@@ -109,10 +112,10 @@ void Mode::get_pilot_desired_steering_and_throttle(float &steering_out, float &t
 }
 
 // decode pilot steering and return steering_out and speed_out (in m/s)
-void Mode::get_pilot_desired_steering_and_speed(float &steering_out, float &speed_out)
+void Mode::get_pilot_desired_steering_and_speed(float &steering_out, float &speed_out, float &lateral_out)
```

Comment:
we never actually use the lateral_out from this function in any modes right now.  I guess the idea is that we will in the future.  We have a "golden rule" in the team that we don't add code before it's actually used because it is sometimes forgotten.

Let's see what others say but maybe we should remove the lateral_out argument from this function.